plugins/USBPrinting/USBPrinterOutputDeviceManager.py

```diff
@@ -128,8 +128,19 @@ def getInstance(cls, engine = None, script_engine = None):
         return USBPrinterOutputDeviceManager._instance
 
     def _getDefaultFirmwareName(self):
-        machine_instance = Application.getInstance().getMachineManager().getActiveMachineInstance()
-        machine_type = machine_instance.getMachineDefinition().getId()
+        # Check whether getGlobalContainerStack() returns None or not...
+        global_container_stack = Application.getInstance().getGlobalContainerStack()
+        if not global_container_stack:
+            Logger.log("c", "getGlobalContainerStack() returned None")
+            Logger.log("i", "Closing firmware-updater UI as a consequence")
+            self._firmware_view.close()
```

Comment:
You should also return after this, otherwise it will still try to update firmware.